consensus/istanbul/types.go

```diff
@@ -26,25 +26,6 @@ import (
 	"github.com/ethereum/go-ethereum/rlp"
 )
 
-// UptimeEntry contains the uptime score of a validator during an epoch as well as the
```

Comment:
Did you move these types because of some circular imports?  If so, what was that circular import?

I actually think that the `Uptime` related types still makes sense to be somewhere in the `istanbul` directory.

Reply:
Yes.
My changes will make `istanbul` depend on `core/state`, because the `BlockProcessResult` has a statedb in `core/state`.
But there is a test in `core/state` depends on `istanbul`

Reply:
Got it.  I actually think that all the uptime related functions and tests should be moved to the `istanbul` package.  But what you did is fine for now.  There are future plans to redo the uptime related code.